modules/EnsEMBL/Web/Component/Gene/ComparaTree.pm

```diff
@@ -50,9 +50,22 @@ sub content {
 
   my $leaves               = $tree->get_all_leaves;
   my $tree_stable_id       = $tree->tree->stable_id;
-  my $highlight_gene       = $self->param('g1');
+  my $highlight_gene       = $hub->param('g1');
+  my $highlight_status     = $hub->get_cookie_value('gene_tree_highlighting') || 'on'; # get the the highlight switch status from the cookie
+
+  # Set $highlight_gene to undefined if the highlight status is off. This is due to the module relying heavily on $highlight_gene to do the rendering based on the highlight status.
```

Comment:
This is duplicated with line 64, you have the same code twice. can you please delete one of them.